### PR TITLE
fix reload systemd configuration after service file change

### PR DESCRIPTION
**Pull request by ghost**
This patch reloads systemd and service configuration after modification of service files.

### CODE REVIEW

**Comment by antoineco** [comment]
Good addition, thanks @scitechfh! Could you add the reload block to _service/archive.pp_ as well to stay consistent?

File: manifests/service/archive.pp [hunk 1]
```diff
@@ -17,29 +17,25 @@
...
-        content => template("${module_name}/instance/systemd_unit_rhel.erb")
-      }
+      $systemd_template = "${module_name}/instance/systemd_unit_suse.erb"
+    } elsif $::operatingsystem == 'Fedora' and $::operatingsystemmajrelease < '20' { # Fedora 17-19
+      $systemd_template = "${module_name}/instance/systemd_unit_fedora.erb"
+    } else { # Fedora 20+ or RHEL 7+
+      $systemd_template = "${module_name}/instance/systemd_unit_rhel.erb"
+    }
+    # write service file
+    file { "${service_name_real} service unit":
```
Comment (antoineco) [inline manifests/service/archive.pp:27]: This variable comes from a higher scope so here you have to use ::tomcat::service_name_real

File: manifests/service/archive.pp [hunk 2]
```diff
@@ -17,29 +17,25 @@
...
+      path    => "/usr/lib/systemd/system/${service_name_real}.service",
```
Comment (antoineco) [inline manifests/service/archive.pp]: Here as well

File: manifests/service/archive.pp [hunk 3]
```diff
@@ -17,29 +17,25 @@
...
+      owner   => 'root',
+      group   => 'root',
+      content => template($systemd_template)
+    }
+    # Refresh systemd configuration
+    exec { "refresh_${service_name_real}":
```
Comment (antoineco) [inline manifests/service/archive.pp:34]: And all over this block too

File: manifests/service/archive.pp (hunk 1 above)
Comment (ghost) [inline manifests/service/archive.pp:27]: I think this is correct. Line 14:
$service_name_real = $::tomcat::service_name_real

Comment (antoineco) [inline manifests/service/archive.pp:27]: You're right, I forgot about that detail. Shame on me.

**Comment by antoineco** [comment]
Thanks again for this :+1:

**Comment by ghost** [comment]
Thanks for merging.

**Pull request merged**